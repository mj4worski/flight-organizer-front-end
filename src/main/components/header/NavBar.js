import React from 'react';
import { Link } from 'react-router-dom';
import cx from 'classnames';
import './NavBar.scss';

const NavItemContainer = ({ children }) => (
  <ul className="nav-bar">
    {React.Children.map(children, child => (
      <li className="nav-bar__item">
        {child}
      </li>
            ))}
  </ul>
);

const NavBar = () => {
  const visualStyles = 'nav-bar__item-visual';
  const selectedStyles = 'nav-bar__item-selected';
  return (
    <NavItemContainer>
      <Link
        to="/"
        className={cx(visualStyles, { [`${selectedStyles}`]: location.pathname === '/' })}
      >
          Strona glowna
      </Link>
      <Link
        to="/findFlight"
        className={cx(visualStyles, { [`${selectedStyles}`]: location.pathname === '/findFlight' })}
      >
          Znajdz lot
      </Link>
      <Link
        to="/login"
        className={cx(visualStyles, { [`${selectedStyles}`]: location.pathname === '/login' })}
      >
          Logowanie
      </Link>
    </NavItemContainer>
  );
};

export default NavBar;
