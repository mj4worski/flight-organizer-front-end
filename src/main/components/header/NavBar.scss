.nav-bar {
  margin: 0;
  padding: 0;
  height: 50px;
  list-style-type: none;
  position: fixed;
  width: 100%;
  display: flex;
  justify-content: space-around;
  background: linear-gradient(to right, #ff9068, #fd746c);

  &__item {
    display: flex;
    flex: 1;
    font-size: 24px;
    font-family: 'Architects Daughter', cursive;
    padding-top: 7px;
    text-align: center;
    &-visual {
      flex: 1;
      text-decoration: none;
      color: white;
    }
    &-visual:hover:not(.active) {
      color: #342d6f;
      opacity: 0.7;
      cursor: pointer;
    }
    &-selected {
      color: #342d6f;
      opacity: 0.7;
    }
  }
}
