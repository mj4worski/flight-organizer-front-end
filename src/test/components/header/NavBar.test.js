import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import NavBar from '../../../main/components/header/NavBar/NavBar';

it('should render NavBar with 3 navigation links', () => {
  const component = renderer.create(
    <MemoryRouter >
      <NavBar />
    </MemoryRouter>,
    );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
