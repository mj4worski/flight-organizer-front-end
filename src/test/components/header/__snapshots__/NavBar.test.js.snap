// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NavBar with 3 navigation links 1`] = `
<ul
  className="nav-bar"
>
  <li
    className="nav-bar__item"
  >
    <a
      className="nav-bar__item-visual"
      href="/"
      onClick={[Function]}
    >
      Main page
    </a>
  </li>
  <li
    className="nav-bar__item"
  >
    <a
      className="nav-bar__item-visual"
      href="/findFlight"
      onClick={[Function]}
    >
      Find flight
    </a>
  </li>
  <li
    className="nav-bar__item"
  >
    <a
      className="nav-bar__item-visual"
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
  </li>
</ul>
`;
